mern/client/src/Components/Navbar/navbar.component.js

```diff
@@ -1,22 +1,53 @@
 import React, { useContext } from "react";
+import { useNavigate } from "react-router-dom";
 import { Link } from "react-router-dom";
 import styles from "./navbar.module.css";
 import GenericButton from "../Buttons/GenericButton";
 import { SessionContext } from "../../Contexts/sessionProvider";
 import { qaContext } from "../../Contexts/qaProvider";
-import deleteFlow from "../../utils/delete";
-import saveFlow from "../../utils/save";
+import { FlowContext } from "../../Contexts/flow.Provider";
+import { deleteFlow } from "../../utils/delete";
+import { saveFlow, saveQA } from "../../utils/save";
 
 export default function Navbar() {
   // define context var to show/hide nav buttons
   const [sessionID, setSessionID] = useContext(SessionContext);
   const [currQA] = useContext(qaContext);
+  const [currFlow, , , , ,] = useContext(FlowContext);
 
   // Create session id var and setter function
   // const sessionid = "12345";
   // const setSessionID = (id) => {
   //   sessionid = id;
   // };
+  const Navigate = useNavigate();
+  const PageChange = (url) => {
+    Navigate(url);
+  };
+
+  // This function is called when user clicks save button and saves the current question
+  const trySave = (currFlow, currQA, sessionID) => {
+    if (!saveQA(currFlow, currQA, sessionID)) {
```

Comment:
Don't you need to do ".then" like how you did for tryDelete?
i thought saveQA was going to return a promise object, similarly with saveFlow

Reply:
Nope. If you check the function, the values in the promises are already extracted.

Reply:
Oh wait nvm, I'm trolling. U right.